pkg/managed/service.go

```diff
@@ -59,11 +66,13 @@ const (
 )
 
 // NewService creates a new Service
-func NewService(provider api.ClusterProvider, stackCollection *manager.StackCollection, clusterName string) *Service {
+func NewService(eksAPI eksiface.EKSAPI, ssmAPI ssmiface.SSMAPI, ec2API ec2iface.EC2API,
+	stackCollection *manager.StackCollection, clusterName string) *Service {
 	return &Service{
```

Comment:
I dislike 5 arguments as it feels like a lot, but at the same time having a new struct to pass the values in by wouldn't make sense as this is a constructor 😂